packages/Webkul/Core/src/Helpers/Exchange/ExchangeRates.php

```diff
@@ -66,11 +74,11 @@ public function updateRates()
 
             if ($exchangeRate = $currency->exchange_rate) {
                 $this->exchangeRateRepository->update([
-                    'rate' => $result['rates'][$currency->code],
+                    'rate' => $result['result'],
                 ], $exchangeRate->id);
             } else {
                 $this->exchangeRateRepository->create([
-                    'rate'            => $result['rates'][$currency->code],
+                    'rate'            => $result['result'],
                     'target_currency' => $currency->id,
                 ]);
             }
```

Comment:
Add a switch case here and check for both two and if the third one or invalid comes up throw an exception (like InvalidExchangeRateException). No need of handling the invalid part.

Reply:
You can use if cases also. But else part should throw an exception.

Reply:
Sorry, ignore the wrong area I added.